### PR TITLE
Introduce wc_is_webhook_valid_status

### LINKED ISSUE #22200: Webhook status update through the API using wrong sanitizing function.
When updating the status of a webhook through the API, the [sanitizing function defined in the schema](https://github.com/woocommerce/woocommerce/blob/e1255de71709d9e7cd8caa143c56fb4d60ab7101/includes/api/v2/class-wc-rest-webhooks-v2-controller.php#L115) is `wc_is_webhook_valid_topic`. This is not a good choice, since a webhook status will never ever be a valid webhook topic.

The sanitizing callback should be a new function `wc_is_webhook_valid_status`, and should check whether the status is valid using [wc_get_webhook_statuses](https://github.com/woocommerce/woocommerce/blob/b0b3ddccb0cf6964e2ff1c70b85dbd3b9764fa85/includes/wc-webhook-functions.php#L98). Unit tests would be nice too. :)

### PR DESCRIPTION
**Pull request by kloon**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:
Webhook status updates via the API was failing to the wrong sanitization callback being used for it. This PR introduces a new function called wc_is_webhook_valid_status and makes use of that function to run as the sanitization callback.

Closes #22200 

### How to test the changes in this Pull Request:

1. Check that new unit tests pass.
2. Make an API call to update a webhook status, check that the status was updated.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### Changelog entry

> Fix - Run webhook status updates through new wc_is_webhook_valid_status functions when doing API requests.

### CODE REVIEW

**Review by claudiosanches — DISMISSED** [review]
Don't need to sanitize, since there is already `enum`, so should only allow those values.
This bug probably generated by copy and paste.

File: includes/api/v2/class-wc-rest-webhooks-v2-controller.php [hunk 1]
```diff
@@ -111,9 +111,6 @@ public function get_item_schema() {
 					'default'     => 'active',
 					'enum'        => array( 'active', 'paused', 'disabled' ),
```
Comment (claudiosanches) [inline includes/api/v2/class-wc-rest-webhooks-v2-controller.php]: You can replace it with `array_keys( wc_get_webhook_statuses() ),`

**Review by claudiulodro — APPROVED** [review]
LGTM. Thanks Gerhard!

**Comment by claudiulodro** [comment]
Just needs merge conflicts resolved then this can be merged.

**Pull request merged**